source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/CustomActionOpenURLRenderer.mm

```diff
@@ -28,12 +28,31 @@ - (UIButton* )renderButton:(UIViewController *)vc
                                              superview:superview
                                      baseActionElement:acoElem
                                             hostConfig:acoConfig];
+    NSString *userSpecifiedBackgroundColor = nil;
     if(button)
     {
-        [button setBackgroundColor:[UIColor colorWithRed:1.0
-                                                   green:59.0/255.0
-                                                    blue:48.0/255.0
-                                                   alpha:1.0]];
+        NSData *additionalProperty = [acoElem additionalProperty];
+        if(additionalProperty) {
+            NSDictionary *dictionary = [NSJSONSerialization JSONObjectWithData:additionalProperty options:NSJSONReadingMutableLeaves error:nil];
+            userSpecifiedBackgroundColor = dictionary[@"color"];
+            NSLog(@"%@", userSpecifiedBackgroundColor);
+        }
+        UIColor *backgroundColor = nil;
+        if(userSpecifiedBackgroundColor) {
+            if([userSpecifiedBackgroundColor isEqualToString: @"green"]) {
```

Comment:
>@"green" [](start = 62, length = 8)

Is this only supposed to work for "green"?

Reply:
yes, i'm just creating an example scenario where users of adaptive cards reads a property and apply its value, and I assume they know mapping of their keys and values. 

---
In reply to: [171957407](https://github.com/Microsoft/AdaptiveCards/pull/1179#discussion_r171957407) [](ancestors = 171957407)